cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -315,7 +315,10 @@ class Date64Converter
  public:
   Status AppendItem(PyObject* obj) {
     int64_t t;
-    if (PyDate_Check(obj)) {
+    if (PyDateTime_Check(obj)) {
```

Comment:
this might not be necessary, if we never expect a datetime object here?

Reply:
In fact I think this is wrong (we always want truncation, but I'll let the reviewer confirm).